minetest/minetest
# Android: Fix settings not saving when closing app via "Recents" screen

Fixes #14229.

On Android, closing the app via the "Recents" screen won't result in a clean shutdown, discarding any setting changes made by the user.

To avoid that, this PR writes the settings file in more cases on Android. It doesn't do this on desktop platforms because you can have multiple instances of Minetest running at the same time there.

An alternative approach would be to ensure a clean shutdown when closing the app from the "Recents" screen, but that's much more complicated (as far as I can tell) and I'd like to fix this bug sooner rather than later.

## To do

This PR is a Ready for Review.

## How to test

Close Minetest on Android from the "Recents" screen. Verify that any setting changes you've made are persisted.
